class-postmarkplugincli.php

```diff
@@ -0,0 +1,3906 @@
+			array_push( $headers, 'From:' . $from );
+
+		}
+
+		// Checks if a subject was specified and uses it.
+		if ( isset( $assoc_args['subject'] ) ) {
+
+			$subject = $assoc_args['subject'];
+
+			// Uses a default subject if not specified.
+		} else {
+
+			$subject = 'Postmark Plugin WP-CLI Test: ' . get_bloginfo( 'name' );
+
+		}
+
+		// Checks if a body was specified and uses it.
+		if ( isset( $assoc_args['body'] ) ) {
+
+			$message = $assoc_args['body'];
+
+			// Uses a default body if not specified.
+		} else {
+
+			$message = 'This is a test email generated from the Postmark for WordPress plugin using the WP-CLI.';
+
+		}
+
+		// Sets open tracking flag.
+		if ( isset( $assoc_args['trackopens'] ) ) {
+
+			$headers['X-PM-Track-Opens'] = true;
+
+		} else {
+
+			$headers['X-PM-Track-Opens'] = false;
+
+		}
+
+		// Sets link tracking flag.
+		if ( isset( $assoc_args['tracklinks'] ) ) {
+
+			$headers['X-PM-Track-Links'] = $assoc_args['tracklinks'];
+
+		} else {
+
+			$headers['X-PM-Track-Links'] = 'None';
+
+		}
+
+		// Sends the test email.
+		$response = wp_mail( $to, $subject, $message, $headers );
+
+		// If all goes well, display a success message using the CLI.
+		if ( false !== $response ) {
+
+			WP_CLI::success(
+				'Successfully sent a test email to ' . $to . '.'
+			);
+		} else {
+
+			$dump = print_r( Postmark_Mail::$last_error, true );
+
+			WP_CLI::warning( 'Test send failed.' );
+
+			WP_CLI::warning( 'Response: ' . $dump );
+		}
+	}
+
+	/**
+	 * Sends a batch of emails.
+	 *
+	 * ## OPTIONS
+	 * <jsonfilename>
+	 * : File containing JSON for each message to send.
+	 *
+	 * [--csv]
+	 * : Output batch results to a csv.
+	 *
+	 * ## EXAMPLES
+	 * $ wp postmark send_batch mybatchdata.json
+	 *
+	 */
+	public function send_batch( $args, $assoc_args ) {
+
+		if ( ! check_server_token_is_set() ) {
+			return;
+		}
+
+		$url = 'https://api.postmarkapp.com/email/batch';
+
+		// Get file contents to use as JSON body.
+		$file = file_get_contents( $args[0], true );
+
+		// Sends batch of emails using the messages in file.
+		$response = $this->postmark_api_call( 'post', $url, $file, null );
+
+		$body = json_decode( $response['body'], true );
+
+		// Successful call
+		if ( 200 === wp_remote_retrieve_response_code( $response ) ) {
+
+			$fields = array( 'Recipient', 'Result', 'Message ID' );
+
+			$data = array();
+
+			foreach ( $body as $send_result ) {
+
+				array_push(
+					$data,
+					array(
+						'Recipient'  => $send_result['To'],
+						'Result'     => $send_result['MessageID'] ? 'Sent' : "Failed - {$send_result['Message']}",
+						'Message ID' => $send_result['MessageID'] ? $send_result['MessageID'] : '',
+					)
+				);
+
+			}
+
+			// Outputs stats in a table.
+			WP_CLI\Utils\format_items( 'table', $data, $fields );
+
+			if ( isset( $assoc_args['csv'] ) ) {
+
+				make_csv( "Recipient, Result, Message ID\n", $body, 'batch_send' );
+
+			}
+
+			// Non 200 API response
+		} else {
+
+			$error_message = [];
```

Comment:
This next block is repeated over and over - might be worth DRYing it up into a method or function.